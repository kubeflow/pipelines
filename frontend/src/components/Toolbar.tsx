/*
 * Copyright 2018 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import IconButton from '@material-ui/core/IconButton';
import Tooltip from '@material-ui/core/Tooltip';
import ArrowBackIcon from '@material-ui/icons/ArrowBack';
import ChevronRightIcon from '@material-ui/icons/ChevronRight';
import { History } from 'history';
import * as React from 'react';
import { CSSProperties } from 'react';
import { Link } from 'react-router-dom';
import { classes, stylesheet } from 'typestyle';
import BusyButton from '../atoms/BusyButton';
import { color, commonCss, dimension, fonts, fontsize, spacing } from '../Css';

export interface ToolbarActionMap {
  [key: string]: ToolbarActionConfig;
}

export interface ToolbarActionConfig {
  action: () => void;
  busy?: boolean;
  disabled?: boolean;
  disabledTitle?: string;
  icon?: any;
  id?: string;
  outlined?: boolean;
  primary?: boolean;
  style?: CSSProperties;
  title: string;
  tooltip: string;
}

export interface Breadcrumb {
  displayName: string;
  href: string;
}

const backIconHeight = 24;

const css = stylesheet({
  actions: {
    display: 'flex',
    marginRight: spacing.units(-2),
  },
  backIcon: {
    fontSize: backIconHeight,
    verticalAlign: 'bottom',
  },
  backLink: {
    cursor: 'pointer',
    marginRight: 10,
    padding: 3,
  },
  breadcrumbs: {
    color: color.inactive,
    fontFamily: fonts.secondary,
    fontSize: fontsize.small,
    letterSpacing: 0.25,
    margin: '10px 37px',
  },
  chevron: {
    height: 12,
  },
  disabled: {
    color: '#aaa',
  },
  enabled: {
    color: color.foreground,
  },
  link: {
    $nest: {
      '&:hover': {
        background: color.lightGrey,
      },
    },
    borderRadius: 3,
    padding: 3,
  },
  pageName: {
    color: color.strong,
    fontSize: fontsize.pageTitle,
    lineHeight: '28px',
  },
  root: {
    alignItems: 'center',
    display: 'flex',
    flexShrink: 0,
    height: dimension.jumbo,
    justifyContent: 'space-between',
  },
  topLevelToolbar: {
    borderBottom: `1px solid ${color.lightGrey}`,
    paddingBottom: 15,
    paddingLeft: 20,
  },
});

export interface ToolbarProps {
  actions: ToolbarActionMap;
  breadcrumbs: Breadcrumb[];
  history?: History;
  pageTitle: string | JSX.Element;
  pageTitleTooltip?: string;
  topLevelToolbar?: boolean;
}

class Toolbar extends React.Component<ToolbarProps> {
  public render(): JSX.Element | null {
    const { actions, breadcrumbs, pageTitle, pageTitleTooltip } = { ...this.props };

    if (!actions.length && !breadcrumbs.length && !pageTitle) {
      return null;
    }

    return (
      <div
        className={classes(css.root, this.props.topLevelToolbar !== false && css.topLevelToolbar)}
      >
        <div style={{ minWidth: 100 }}>
          {/* Breadcrumb */}
          <div className={classes(css.breadcrumbs, commonCss.flex)}>
            {breadcrumbs.map((crumb, i) => (
              <span className={commonCss.flex} key={i} title={crumb.displayName}>
                {i !== 0 && <ChevronRightIcon className={css.chevron} />}
                <Link
                  className={classes(commonCss.unstyled, commonCss.ellipsis, css.link)}
                  to={crumb.href}
                >
                  {crumb.displayName}
                </Link>
              </span>
            ))}
          </div>
          <div className={commonCss.flex}>
            {/* Back Arrow */}
            {breadcrumbs.length > 0 && (
              <Tooltip title={'Back'} enterDelay={300}>
                <div>
                  {' '}
                  {/* Div needed because we sometimes disable a button within a tooltip */}
                  <IconButton
                    className={css.backLink}
                    disabled={this.props.history!.length < 2}
                    onClick={this.props.history!.goBack}
                  >
                    <ArrowBackIcon
                      className={classes(
                        css.backIcon,
                        this.props.history!.length < 2 ? css.disabled : css.enabled,
                      )}
                    />
                  </IconButton>
                </div>
              </Tooltip>
            )}
            {/* Resource Name */}
            <span
              className={classes(css.pageName, commonCss.ellipsis)}
              title={pageTitleTooltip}
              data-testid='page-title' // TODO: use a proper h1 tag for page titles and let tests query this by h1.
            >
              {pageTitle}
            </span>
          </div>
        </div>
        {/* Actions / Buttons */}
        <div className={css.actions}>
          {Object.keys(actions).map((buttonKey, i) => {
            const button = actions[buttonKey];
            return (
              <Tooltip
                title={
                  button.disabled && button.disabledTitle ? button.disabledTitle : button.tooltip
                }
                enterDelay={300}
                key={i}
              >
                <div style={button.style}>
                  {/* Extra level needed by tooltip when child is disabled */}
                  <BusyButton
                    id={button.id}
                    color='secondary'
                    onClick={button.action}
                    disabled={button.disabled}
                    title={button.title}
                    icon={button.icon}
                    busy={button.busy || false}
                    outlined={(button.outlined && !button.primary) || false}
                    className={button.primary ? commonCss.buttonAction : ''}
                  />
                </div>
              </Tooltip>
            );
          })}
        </div>
      </div>
    );
  }
}

export default Toolbar;
