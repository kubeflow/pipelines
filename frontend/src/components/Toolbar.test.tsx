/*
 * Copyright 2018 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import * as React from 'react';
import { shallow } from 'enzyme';
import { createBrowserHistory, createMemoryHistory } from 'history';
import Toolbar, { ToolbarActionMap } from './Toolbar';
import HelpIcon from '@material-ui/icons/Help';
import InfoIcon from '@material-ui/icons/Info';

const action1 = jest.fn();
const action2 = jest.fn();
const actions: ToolbarActionMap = {
  action1: {
    action: action1,
    disabledTitle: 'test disabled title',
    icon: HelpIcon,
    id: 'test id',
    title: 'test title',
    tooltip: 'test tooltip',
  },
  action2: {
    action: action2,
    disabled: true,
    disabledTitle: 'test disabled title2',
    icon: InfoIcon,
    id: 'test id2',
    title: 'test title2',
    tooltip: 'test tooltip2',
  },
};

const breadcrumbs = [
  {
    displayName: 'test display name',
    href: '/some/test/path',
  },
  {
    displayName: 'test display name2',
    href: '/some/test/path2',
  },
];

const history = createBrowserHistory({});

describe('Toolbar', () => {
  beforeAll(() => {
    history.push('/pipelines');
  });

  it('renders nothing when there are no breadcrumbs or actions', () => {
    const tree = shallow(<Toolbar breadcrumbs={[]} actions={{}} history={history} pageTitle='' />);
    expect(tree).toMatchSnapshot();
  });

  it('renders without breadcrumbs and a string page title', () => {
    const tree = shallow(
      <Toolbar breadcrumbs={[]} actions={actions} history={history} pageTitle='test page title' />,
    );
    expect(tree).toMatchSnapshot();
  });

  it('renders without breadcrumbs and a component page title', () => {
    const tree = shallow(
      <Toolbar
        breadcrumbs={[]}
        actions={actions}
        history={history}
        pageTitle={<div id='myComponent'>test page title</div>}
      />,
    );
    expect(tree).toMatchSnapshot();
  });

  it('renders without breadcrumbs and one action', () => {
    const singleAction = {
      action1: {
        action: action1,
        disabledTitle: 'test disabled title',
        icon: HelpIcon,
        id: 'test id',
        title: 'test title',
        tooltip: 'test tooltip',
      },
    };
    const tree = shallow(
      <Toolbar
        breadcrumbs={[]}
        actions={singleAction}
        history={history}
        pageTitle='test page title'
      />,
    );
    expect(tree).toMatchSnapshot();
  });

  it('renders without actions and one breadcrumb', () => {
    const tree = shallow(
      <Toolbar
        breadcrumbs={[breadcrumbs[0]]}
        actions={{}}
        history={history}
        pageTitle='test page title'
      />,
    );
    expect(tree).toMatchSnapshot();
  });

  it('renders without actions, one breadcrumb, and a page name', () => {
    const tree = shallow(
      <Toolbar
        breadcrumbs={[breadcrumbs[0]]}
        actions={{}}
        history={history}
        pageTitle='test page title'
      />,
    );
    expect(tree).toMatchSnapshot();
  });

  it('renders without breadcrumbs and two actions', () => {
    const tree = shallow(
      <Toolbar breadcrumbs={[]} actions={actions} history={history} pageTitle='test page title' />,
    );
    expect(tree).toMatchSnapshot();
  });

  it('fires the right action function when button is clicked', () => {
    const tree = shallow(
      <Toolbar breadcrumbs={[]} actions={actions} history={history} pageTitle='test page title' />,
    );
    tree
      .find('BusyButton')
      .at(0)
      .simulate('click');
    expect(action1).toHaveBeenCalled();
    action1.mockClear();
  });

  it('renders outlined action buttons', () => {
    const outlinedActions = {
      action1: {
        action: jest.fn(),
        id: 'test outlined id',
        outlined: true,
        title: 'test outlined title',
        tooltip: 'test outlined tooltip',
      },
    };

    const tree = shallow(
      <Toolbar
        breadcrumbs={breadcrumbs}
        actions={outlinedActions}
        pageTitle=''
        history={history}
      />,
    );
    expect(tree).toMatchSnapshot();
  });

  it('renders primary action buttons', () => {
    const primaryActions = {
      action1: {
        action: jest.fn(),
        id: 'test primary id',
        primary: true,
        title: 'test primary title',
        tooltip: 'test primary tooltip',
      },
    };

    const tree = shallow(
      <Toolbar breadcrumbs={breadcrumbs} actions={primaryActions} pageTitle='' history={history} />,
    );
    expect(tree).toMatchSnapshot();
  });

  it('renders primary action buttons without outline, even if outline is true', () => {
    const outlinedPrimaryActions = {
      action1: {
        action: jest.fn(),
        id: 'test id',
        outlined: true,
        primary: true,
        title: 'test title',
        tooltip: 'test tooltip',
      },
    };

    const tree = shallow(
      <Toolbar
        breadcrumbs={breadcrumbs}
        actions={outlinedPrimaryActions}
        pageTitle=''
        history={history}
      />,
    );
    expect(tree).toMatchSnapshot();
  });

  it('renders with two breadcrumbs and two actions', () => {
    const tree = shallow(
      <Toolbar breadcrumbs={breadcrumbs} actions={actions} pageTitle='' history={history} />,
    );
    expect(tree).toMatchSnapshot();
  });

  it('disables the back button when there is no browser history', () => {
    // This test uses createMemoryHistory because createBroweserHistory returns a singleton, and
    // there is no way to clear its entries which this test requires.
    const emptyHistory = createMemoryHistory();
    const tree = shallow(
      <Toolbar breadcrumbs={breadcrumbs} actions={actions} history={emptyHistory} pageTitle='' />,
    );
    expect(tree).toMatchSnapshot();
  });
});
