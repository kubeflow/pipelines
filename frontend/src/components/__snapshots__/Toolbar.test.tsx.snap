// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toolbar disables the back button when there is no browser history 1`] = `
<div
  className="root topLevelToolbar"
>
  <div
    style={
      Object {
        "minWidth": 100,
      }
    }
  >
    <div
      className="breadcrumbs flex"
    >
      <span
        className="flex"
        key="0"
        title="test display name"
      >
        <Link
          className="unstyled ellipsis link"
          replace={false}
          to="/some/test/path"
        >
          test display name
        </Link>
      </span>
      <span
        className="flex"
        key="1"
        title="test display name2"
      >
        <pure(ChevronRightIcon)
          className="chevron"
        />
        <Link
          className="unstyled ellipsis link"
          replace={false}
          to="/some/test/path2"
        >
          test display name2
        </Link>
      </span>
    </div>
    <div
      className="flex"
    >
      <WithStyles(Tooltip)
        enterDelay={300}
        title="Back"
      >
        <div>
           
          <WithStyles(IconButton)
            className="backLink"
            disabled={true}
            onClick={[Function]}
          >
            <pure(ArrowBackIcon)
              className="backIcon disabled"
            />
          </WithStyles(IconButton)>
        </div>
      </WithStyles(Tooltip)>
      <span
        className="pageName ellipsis"
        data-testid="page-title"
      />
    </div>
  </div>
  <div
    className="actions"
  >
    <WithStyles(Tooltip)
      enterDelay={300}
      key="0"
      title="test tooltip"
    >
      <div>
        <BusyButton
          busy={false}
          className=""
          color="secondary"
          icon={[Function]}
          id="test id"
          onClick={[MockFunction]}
          outlined={false}
          title="test title"
        />
      </div>
    </WithStyles(Tooltip)>
    <WithStyles(Tooltip)
      enterDelay={300}
      key="1"
      title="test disabled title2"
    >
      <div>
        <BusyButton
          busy={false}
          className=""
          color="secondary"
          disabled={true}
          icon={[Function]}
          id="test id2"
          onClick={[MockFunction]}
          outlined={false}
          title="test title2"
        />
      </div>
    </WithStyles(Tooltip)>
  </div>
</div>
`;

exports[`Toolbar renders nothing when there are no breadcrumbs or actions 1`] = `""`;

exports[`Toolbar renders outlined action buttons 1`] = `
<div
  className="root topLevelToolbar"
>
  <div
    style={
      Object {
        "minWidth": 100,
      }
    }
  >
    <div
      className="breadcrumbs flex"
    >
      <span
        className="flex"
        key="0"
        title="test display name"
      >
        <Link
          className="unstyled ellipsis link"
          replace={false}
          to="/some/test/path"
        >
          test display name
        </Link>
      </span>
      <span
        className="flex"
        key="1"
        title="test display name2"
      >
        <pure(ChevronRightIcon)
          className="chevron"
        />
        <Link
          className="unstyled ellipsis link"
          replace={false}
          to="/some/test/path2"
        >
          test display name2
        </Link>
      </span>
    </div>
    <div
      className="flex"
    >
      <WithStyles(Tooltip)
        enterDelay={300}
        title="Back"
      >
        <div>
           
          <WithStyles(IconButton)
            className="backLink"
            disabled={false}
            onClick={[Function]}
          >
            <pure(ArrowBackIcon)
              className="backIcon enabled"
            />
          </WithStyles(IconButton)>
        </div>
      </WithStyles(Tooltip)>
      <span
        className="pageName ellipsis"
        data-testid="page-title"
      />
    </div>
  </div>
  <div
    className="actions"
  >
    <WithStyles(Tooltip)
      enterDelay={300}
      key="0"
      title="test outlined tooltip"
    >
      <div>
        <BusyButton
          busy={false}
          className=""
          color="secondary"
          id="test outlined id"
          onClick={[MockFunction]}
          outlined={true}
          title="test outlined title"
        />
      </div>
    </WithStyles(Tooltip)>
  </div>
</div>
`;

exports[`Toolbar renders primary action buttons 1`] = `
<div
  className="root topLevelToolbar"
>
  <div
    style={
      Object {
        "minWidth": 100,
      }
    }
  >
    <div
      className="breadcrumbs flex"
    >
      <span
        className="flex"
        key="0"
        title="test display name"
      >
        <Link
          className="unstyled ellipsis link"
          replace={false}
          to="/some/test/path"
        >
          test display name
        </Link>
      </span>
      <span
        className="flex"
        key="1"
        title="test display name2"
      >
        <pure(ChevronRightIcon)
          className="chevron"
        />
        <Link
          className="unstyled ellipsis link"
          replace={false}
          to="/some/test/path2"
        >
          test display name2
        </Link>
      </span>
    </div>
    <div
      className="flex"
    >
      <WithStyles(Tooltip)
        enterDelay={300}
        title="Back"
      >
        <div>
           
          <WithStyles(IconButton)
            className="backLink"
            disabled={false}
            onClick={[Function]}
          >
            <pure(ArrowBackIcon)
              className="backIcon enabled"
            />
          </WithStyles(IconButton)>
        </div>
      </WithStyles(Tooltip)>
      <span
        className="pageName ellipsis"
        data-testid="page-title"
      />
    </div>
  </div>
  <div
    className="actions"
  >
    <WithStyles(Tooltip)
      enterDelay={300}
      key="0"
      title="test primary tooltip"
    >
      <div>
        <BusyButton
          busy={false}
          className="buttonAction"
          color="secondary"
          id="test primary id"
          onClick={[MockFunction]}
          outlined={false}
          title="test primary title"
        />
      </div>
    </WithStyles(Tooltip)>
  </div>
</div>
`;

exports[`Toolbar renders primary action buttons without outline, even if outline is true 1`] = `
<div
  className="root topLevelToolbar"
>
  <div
    style={
      Object {
        "minWidth": 100,
      }
    }
  >
    <div
      className="breadcrumbs flex"
    >
      <span
        className="flex"
        key="0"
        title="test display name"
      >
        <Link
          className="unstyled ellipsis link"
          replace={false}
          to="/some/test/path"
        >
          test display name
        </Link>
      </span>
      <span
        className="flex"
        key="1"
        title="test display name2"
      >
        <pure(ChevronRightIcon)
          className="chevron"
        />
        <Link
          className="unstyled ellipsis link"
          replace={false}
          to="/some/test/path2"
        >
          test display name2
        </Link>
      </span>
    </div>
    <div
      className="flex"
    >
      <WithStyles(Tooltip)
        enterDelay={300}
        title="Back"
      >
        <div>
           
          <WithStyles(IconButton)
            className="backLink"
            disabled={false}
            onClick={[Function]}
          >
            <pure(ArrowBackIcon)
              className="backIcon enabled"
            />
          </WithStyles(IconButton)>
        </div>
      </WithStyles(Tooltip)>
      <span
        className="pageName ellipsis"
        data-testid="page-title"
      />
    </div>
  </div>
  <div
    className="actions"
  >
    <WithStyles(Tooltip)
      enterDelay={300}
      key="0"
      title="test tooltip"
    >
      <div>
        <BusyButton
          busy={false}
          className="buttonAction"
          color="secondary"
          id="test id"
          onClick={[MockFunction]}
          outlined={false}
          title="test title"
        />
      </div>
    </WithStyles(Tooltip)>
  </div>
</div>
`;

exports[`Toolbar renders with two breadcrumbs and two actions 1`] = `
<div
  className="root topLevelToolbar"
>
  <div
    style={
      Object {
        "minWidth": 100,
      }
    }
  >
    <div
      className="breadcrumbs flex"
    >
      <span
        className="flex"
        key="0"
        title="test display name"
      >
        <Link
          className="unstyled ellipsis link"
          replace={false}
          to="/some/test/path"
        >
          test display name
        </Link>
      </span>
      <span
        className="flex"
        key="1"
        title="test display name2"
      >
        <pure(ChevronRightIcon)
          className="chevron"
        />
        <Link
          className="unstyled ellipsis link"
          replace={false}
          to="/some/test/path2"
        >
          test display name2
        </Link>
      </span>
    </div>
    <div
      className="flex"
    >
      <WithStyles(Tooltip)
        enterDelay={300}
        title="Back"
      >
        <div>
           
          <WithStyles(IconButton)
            className="backLink"
            disabled={false}
            onClick={[Function]}
          >
            <pure(ArrowBackIcon)
              className="backIcon enabled"
            />
          </WithStyles(IconButton)>
        </div>
      </WithStyles(Tooltip)>
      <span
        className="pageName ellipsis"
        data-testid="page-title"
      />
    </div>
  </div>
  <div
    className="actions"
  >
    <WithStyles(Tooltip)
      enterDelay={300}
      key="0"
      title="test tooltip"
    >
      <div>
        <BusyButton
          busy={false}
          className=""
          color="secondary"
          icon={[Function]}
          id="test id"
          onClick={[MockFunction]}
          outlined={false}
          title="test title"
        />
      </div>
    </WithStyles(Tooltip)>
    <WithStyles(Tooltip)
      enterDelay={300}
      key="1"
      title="test disabled title2"
    >
      <div>
        <BusyButton
          busy={false}
          className=""
          color="secondary"
          disabled={true}
          icon={[Function]}
          id="test id2"
          onClick={[MockFunction]}
          outlined={false}
          title="test title2"
        />
      </div>
    </WithStyles(Tooltip)>
  </div>
</div>
`;

exports[`Toolbar renders without actions and one breadcrumb 1`] = `
<div
  className="root topLevelToolbar"
>
  <div
    style={
      Object {
        "minWidth": 100,
      }
    }
  >
    <div
      className="breadcrumbs flex"
    >
      <span
        className="flex"
        key="0"
        title="test display name"
      >
        <Link
          className="unstyled ellipsis link"
          replace={false}
          to="/some/test/path"
        >
          test display name
        </Link>
      </span>
    </div>
    <div
      className="flex"
    >
      <WithStyles(Tooltip)
        enterDelay={300}
        title="Back"
      >
        <div>
           
          <WithStyles(IconButton)
            className="backLink"
            disabled={false}
            onClick={[Function]}
          >
            <pure(ArrowBackIcon)
              className="backIcon enabled"
            />
          </WithStyles(IconButton)>
        </div>
      </WithStyles(Tooltip)>
      <span
        className="pageName ellipsis"
        data-testid="page-title"
      >
        test page title
      </span>
    </div>
  </div>
  <div
    className="actions"
  />
</div>
`;

exports[`Toolbar renders without actions, one breadcrumb, and a page name 1`] = `
<div
  className="root topLevelToolbar"
>
  <div
    style={
      Object {
        "minWidth": 100,
      }
    }
  >
    <div
      className="breadcrumbs flex"
    >
      <span
        className="flex"
        key="0"
        title="test display name"
      >
        <Link
          className="unstyled ellipsis link"
          replace={false}
          to="/some/test/path"
        >
          test display name
        </Link>
      </span>
    </div>
    <div
      className="flex"
    >
      <WithStyles(Tooltip)
        enterDelay={300}
        title="Back"
      >
        <div>
           
          <WithStyles(IconButton)
            className="backLink"
            disabled={false}
            onClick={[Function]}
          >
            <pure(ArrowBackIcon)
              className="backIcon enabled"
            />
          </WithStyles(IconButton)>
        </div>
      </WithStyles(Tooltip)>
      <span
        className="pageName ellipsis"
        data-testid="page-title"
      >
        test page title
      </span>
    </div>
  </div>
  <div
    className="actions"
  />
</div>
`;

exports[`Toolbar renders without breadcrumbs and a component page title 1`] = `
<div
  className="root topLevelToolbar"
>
  <div
    style={
      Object {
        "minWidth": 100,
      }
    }
  >
    <div
      className="breadcrumbs flex"
    />
    <div
      className="flex"
    >
      <span
        className="pageName ellipsis"
        data-testid="page-title"
      >
        <div
          id="myComponent"
        >
          test page title
        </div>
      </span>
    </div>
  </div>
  <div
    className="actions"
  >
    <WithStyles(Tooltip)
      enterDelay={300}
      key="0"
      title="test tooltip"
    >
      <div>
        <BusyButton
          busy={false}
          className=""
          color="secondary"
          icon={[Function]}
          id="test id"
          onClick={[MockFunction]}
          outlined={false}
          title="test title"
        />
      </div>
    </WithStyles(Tooltip)>
    <WithStyles(Tooltip)
      enterDelay={300}
      key="1"
      title="test disabled title2"
    >
      <div>
        <BusyButton
          busy={false}
          className=""
          color="secondary"
          disabled={true}
          icon={[Function]}
          id="test id2"
          onClick={[MockFunction]}
          outlined={false}
          title="test title2"
        />
      </div>
    </WithStyles(Tooltip)>
  </div>
</div>
`;

exports[`Toolbar renders without breadcrumbs and a string page title 1`] = `
<div
  className="root topLevelToolbar"
>
  <div
    style={
      Object {
        "minWidth": 100,
      }
    }
  >
    <div
      className="breadcrumbs flex"
    />
    <div
      className="flex"
    >
      <span
        className="pageName ellipsis"
        data-testid="page-title"
      >
        test page title
      </span>
    </div>
  </div>
  <div
    className="actions"
  >
    <WithStyles(Tooltip)
      enterDelay={300}
      key="0"
      title="test tooltip"
    >
      <div>
        <BusyButton
          busy={false}
          className=""
          color="secondary"
          icon={[Function]}
          id="test id"
          onClick={[MockFunction]}
          outlined={false}
          title="test title"
        />
      </div>
    </WithStyles(Tooltip)>
    <WithStyles(Tooltip)
      enterDelay={300}
      key="1"
      title="test disabled title2"
    >
      <div>
        <BusyButton
          busy={false}
          className=""
          color="secondary"
          disabled={true}
          icon={[Function]}
          id="test id2"
          onClick={[MockFunction]}
          outlined={false}
          title="test title2"
        />
      </div>
    </WithStyles(Tooltip)>
  </div>
</div>
`;

exports[`Toolbar renders without breadcrumbs and one action 1`] = `
<div
  className="root topLevelToolbar"
>
  <div
    style={
      Object {
        "minWidth": 100,
      }
    }
  >
    <div
      className="breadcrumbs flex"
    />
    <div
      className="flex"
    >
      <span
        className="pageName ellipsis"
        data-testid="page-title"
      >
        test page title
      </span>
    </div>
  </div>
  <div
    className="actions"
  >
    <WithStyles(Tooltip)
      enterDelay={300}
      key="0"
      title="test tooltip"
    >
      <div>
        <BusyButton
          busy={false}
          className=""
          color="secondary"
          icon={[Function]}
          id="test id"
          onClick={[MockFunction]}
          outlined={false}
          title="test title"
        />
      </div>
    </WithStyles(Tooltip)>
  </div>
</div>
`;

exports[`Toolbar renders without breadcrumbs and two actions 1`] = `
<div
  className="root topLevelToolbar"
>
  <div
    style={
      Object {
        "minWidth": 100,
      }
    }
  >
    <div
      className="breadcrumbs flex"
    />
    <div
      className="flex"
    >
      <span
        className="pageName ellipsis"
        data-testid="page-title"
      >
        test page title
      </span>
    </div>
  </div>
  <div
    className="actions"
  >
    <WithStyles(Tooltip)
      enterDelay={300}
      key="0"
      title="test tooltip"
    >
      <div>
        <BusyButton
          busy={false}
          className=""
          color="secondary"
          icon={[Function]}
          id="test id"
          onClick={[MockFunction]}
          outlined={false}
          title="test title"
        />
      </div>
    </WithStyles(Tooltip)>
    <WithStyles(Tooltip)
      enterDelay={300}
      key="1"
      title="test disabled title2"
    >
      <div>
        <BusyButton
          busy={false}
          className=""
          color="secondary"
          disabled={true}
          icon={[Function]}
          id="test id2"
          onClick={[MockFunction]}
          outlined={false}
          title="test title2"
        />
      </div>
    </WithStyles(Tooltip)>
  </div>
</div>
`;
